pkg/executables/kubectl_test.go

```diff
@@ -2836,3 +2836,77 @@ func TestKubectlDeleteEksaNutanixMachineConfigError(t *testing.T) {
 	).Return(bytes.Buffer{}, fmt.Errorf("error"))
 	tt.Expect(tt.k.DeleteEksaNutanixMachineConfig(tt.ctx, "eksa-unit-test", tt.kubeconfig, tt.namespace)).NotTo(Succeed())
 }
+
+func TestKubectlSearchTinkerbellMachineConfig(t *testing.T) {
+	var kubeconfig, namespace, name string
+	buffer := bytes.Buffer{}
+	buffer.WriteString(test.ReadFile(t, "testdata/kubectl_no_cs_machineconfigs.json"))
+	k, ctx, _, e := newKubectl(t)
+
+	expectedParam := []string{
+		"get", fmt.Sprintf("tinkerbellmachineconfigs.%s", v1alpha1.GroupVersion.Group), "-o", "json", "--kubeconfig",
+		kubeconfig, "--namespace", namespace, "--field-selector=metadata.name=" + name,
+	}
+	e.EXPECT().Execute(ctx, gomock.Eq(expectedParam)).Return(buffer, nil)
+	mc, err := k.SearchTinkerbellMachineConfig(ctx, name, kubeconfig, namespace)
+	if err != nil {
+		t.Errorf("Kubectl.SearchTinkerbellMachineConfig() error = %v, want nil", err)
+	}
+	if len(mc) > 0 {
+		t.Errorf("expected 0 machine configs, got %d", len(mc))
+	}
+}
+
+func TestKubectlSearchTinkerbellMachineConfigInvalid(t *testing.T) {
+	var kubeconfig, namespace, name string
+	machineconfigJSON := test.ReadFile(t, "testdata/kubectl_tinkerbellmachineconfig_invalid.json")
```

Comment:
We go to the effort of loading a test file but we don't seem to do any checks to ensure we got the content we expected.